setup.py

```diff
@@ -16,7 +16,7 @@
 
 extras = {
     'dev': ['bump2version'],
-    'opt': ['oapackage', 'pqkmeans'],
+    'opt': ['oapackage', 'pqkmeans', 'openpmd-api'],
```

Comment:
If a dependency (an optional one in this case) is added without updating the Conda environment accordingly at the same time, the test pipeline fails as it checks whether the provided environment meets all the specified requirements of MALA. 

In the CPU-test pipeline MALA gets installed with optional and test dependencies (as specified in  `setup.py`):

https://github.com/mala-project/mala/blob/2d8c5217ec83070a56c63b96a456296dfe9853f2/.github/workflows/cpu-tests.yml#L158

Since `openpmd` is not (yet) part of the Conda environment within the Docker container, it is simply installed additionally and thus the deviation from the given environment is detected.

![image](https://user-images.githubusercontent.com/28047702/208946107-440e75d9-622c-4f33-8237-9deaa86c4dbe.png)

As we want to provide an environment that fully meets the requirements of MALA, and also to avoid installation time of packages in CI jobs, such a deviation is detected and aborts the pipeline correctly. As a consequence this PR should not have been merged in the first place without fixing the Conda environment in the Docker container. This is now being resolved retrospectively in PR #383.

Reply:
Understood! I will add a chapter in the documentation "contributing" section on how to properly handle the addition of such dependencies.